src/app/offers/offer-details/offer-details.component.ts

```diff
@@ -25,7 +25,9 @@ export class OfferDetailsComponent implements OnInit {
     isOverview = false;
     isPortfolio = false;
     userConfirmed = true;
-    currentLocation = encodeURIComponent(window.location.href);
+    projectID = this.route.snapshot.params.id;
```

Comment:
This cannot be inferred when the component is created! 

Snapshot should be _at the earliest_ inferred in `ngOnInit()` - take a look at the component lifecycle.

In general **don't work in declarations** - work in `constructor` or `ngOnInit` or other methods. Declarations should never be used for time sensitive stuff.

The order of operations might be completely wrong and how would you know?

Reply:
Study this https://angular.io/guide/lifecycle-hooks#lifecycle-event-sequence